<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GeneratorTab</class>
 <widget class="QWidget" name="GeneratorTab">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1011</width>
    <height>669</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>1160</width>
        <height>638</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QSplitter" name="splitter">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <widget class="QFrame" name="frame_2">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QTabWidget" name="tabWidget">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="tabPosition">
              <enum>QTabWidget::North</enum>
             </property>
             <property name="tabShape">
              <enum>QTabWidget::Rounded</enum>
             </property>
             <property name="currentIndex">
              <number>0</number>
             </property>
             <widget class="QWidget" name="tab_proto">
              <attribute name="title">
               <string>Protocols</string>
              </attribute>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="GeneratorTreeView" name="treeProtocols">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="toolTip">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Drag&amp;amp;Drop Protocols to the table on the right to fill the generation table.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <attribute name="headerDefaultSectionSize">
                  <number>0</number>
                 </attribute>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tab_4">
              <attribute name="title">
               <string>Pauses</string>
              </attribute>
              <layout class="QGridLayout" name="gridLayout_4">
               <item row="0" column="0" colspan="2">
                <widget class="GeneratorListWidget" name="lWPauses">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="toolTip">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The pauses will be added automatically when you drag a protocol from the tree above to the table on the right.&lt;br/&gt;&lt;/p&gt;&lt;p&gt;You can see the &lt;span style=&quot; font-weight:600;&quot;&gt;position&lt;/span&gt; of each pause by &lt;span style=&quot; font-weight:600;&quot;&gt;selecting it&lt;/span&gt;. There will be drawn a line in the table indiciating the position of the pause.&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Use context menu or double click to &lt;span style=&quot; font-weight:600;&quot;&gt;edit a pauses' length&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="editTriggers">
                  <set>QAbstractItemView::DoubleClicked|QAbstractItemView::EditKeyPressed</set>
                 </property>
                 <property name="showDropIndicator" stdset="0">
                  <bool>false</bool>
                 </property>
                 <property name="dragDropMode">
                  <enum>QAbstractItemView::NoDragDrop</enum>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tab_5">
              <attribute name="title">
               <string>Fuzzing</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_8">
               <property name="spacing">
                <number>6</number>
               </property>
               <item>
                <widget class="GeneratorListView" name="listViewProtoLabels">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="editTriggers">
                  <set>QAbstractItemView::EditKeyPressed</set>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QGroupBox" name="groupBox">
                 <property name="title">
                  <string>Add fuzzing values to generated data</string>
                 </property>
                 <layout class="QHBoxLayout" name="horizontalLayout_5">
                  <item>
                   <widget class="QStackedWidget" name="stackedWidgetFuzzing">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                    <property name="currentIndex">
                     <number>0</number>
                    </property>
                    <widget class="QWidget" name="pageFuzzingUI">
                     <layout class="QHBoxLayout" name="horizontalLayout_2">
                      <item>
                       <widget class="QPushButton" name="btnFuzz">
                        <property name="text">
                         <string>Fuzz</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QRadioButton" name="rBSuccessive">
                        <property name="toolTip">
                         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;For multiple labels per message the fuzzed values are inserted &lt;span style=&quot; font-weight:600;&quot;&gt;one-by-one&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                        </property>
                        <property name="text">
                         <string>S&amp;uccessive</string>
                        </property>
                        <property name="checked">
                         <bool>true</bool>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QRadioButton" name="rbConcurrent">
                        <property name="toolTip">
                         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;For multiple labels per message the labels are fuzzed &lt;span style=&quot; font-weight:600;&quot;&gt;at the same time&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                        </property>
                        <property name="text">
                         <string>&amp;Concurrent</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QRadioButton" name="rBExhaustive">
                        <property name="toolTip">
                         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;For multiple labels per message the fuzzed values are inserted in &lt;span style=&quot; font-weight:600;&quot;&gt;all possible combinations&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                        </property>
                        <property name="text">
                         <string>E&amp;xhaustive</string>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </widget>
                    <widget class="QWidget" name="pageFuzzingProgressBar">
                     <layout class="QHBoxLayout" name="horizontalLayout_3">
                      <item>
                       <widget class="QProgressBar" name="progressBarFuzzing">
                        <property name="value">
                         <number>24</number>
                        </property>
                        <property name="format">
                         <string>%v/%m</string>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </widget>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="2" column="1">
              <widget class="QComboBox" name="cBoxModulations">
               <item>
                <property name="text">
                 <string>MyModulation</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="0" column="0" colspan="3">
              <layout class="QGridLayout" name="modulationLayout">
               <item row="0" column="0">
                <widget class="QLabel" name="lEncoding">
                 <property name="text">
                  <string>Encoding:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLabel" name="lEncodingValue">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>-</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QLabel" name="lSampleRate">
                 <property name="text">
                  <string>Sample Rate:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="3">
                <widget class="QLabel" name="lSampleRateValue">
                 <property name="text">
                  <string>TextLabel</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="lCarrierFrequency">
                 <property name="text">
                  <string>Carrier Frequency:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLabel" name="lCarrierFreqValue">
                 <property name="text">
                  <string>TextLabel</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <widget class="QLabel" name="lModType">
                 <property name="text">
                  <string>Modulation Type:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="3">
                <widget class="QLabel" name="lModTypeValue">
                 <property name="text">
                  <string>TextLabel</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>Carrier Phase:</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QLabel" name="lCarrierPhaseValue">
                 <property name="text">
                  <string>TextLabel</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="2">
                <widget class="QLabel" name="lParamForZero">
                 <property name="text">
                  <string>Amplitude for 0:</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="3">
                <widget class="QLabel" name="lParamForZeroValue">
                 <property name="text">
                  <string>TextLabel</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="lBitLength">
                 <property name="text">
                  <string>Bit Length:</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QLabel" name="lBitLenValue">
                 <property name="text">
                  <string>TextLabel</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="2">
                <widget class="QLabel" name="lParamForOne">
                 <property name="text">
                  <string>Amplitude for 1:</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="3">
                <widget class="QLabel" name="lParamForOneValue">
                 <property name="text">
                  <string>TextLabel</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="lModulation">
               <property name="text">
                <string>Modulation:</string>
               </property>
              </widget>
             </item>
             <item row="5" column="2">
              <widget class="QPushButton" name="btnSend">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Send data...</string>
               </property>
               <property name="icon">
                <iconset theme="media-playback-start">
                 <normaloff>.</normaloff>.</iconset>
               </property>
              </widget>
             </item>
             <item row="1" column="0" colspan="3">
              <widget class="Line" name="line">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="2" column="2">
              <widget class="QPushButton" name="btnEditModulation">
               <property name="text">
                <string>Edit ...</string>
               </property>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QProgressBar" name="prBarGeneration">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="value">
                <number>0</number>
               </property>
               <property name="format">
                <string>Modulating %p%</string>
               </property>
              </widget>
             </item>
             <item row="5" column="1">
              <widget class="QPushButton" name="btnGenerate">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="toolTip">
                <string>Generate the complex file of the modulated signal, after tuning all parameters above.</string>
               </property>
               <property name="text">
                <string>Generate file...</string>
               </property>
               <property name="icon">
                <iconset theme="document-new"/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QFrame" name="frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="2" column="7">
            <widget class="QLabel" name="lViewType">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Viewtype:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="6">
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>38</width>
               <height>22</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="2" column="3">
            <widget class="QLabel" name="lEstimatedTime">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The estimated average time is based on the average number of bits per message and average sample rate, you set for the modulations.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>Estimated Time: </string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QToolButton" name="btnSave">
             <property name="text">
              <string>...</string>
             </property>
             <property name="icon">
              <iconset theme="document-save">
               <normaloff>.</normaloff>.</iconset>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="9">
            <widget class="GeneratorTableView" name="tableMessages">
             <property name="acceptDrops">
              <bool>true</bool>
             </property>
             <property name="dragEnabled">
              <bool>false</bool>
             </property>
             <property name="dragDropOverwriteMode">
              <bool>false</bool>
             </property>
             <property name="dragDropMode">
              <enum>QAbstractItemView::DropOnly</enum>
             </property>
             <property name="defaultDropAction">
              <enum>Qt::CopyAction</enum>
             </property>
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="selectionBehavior">
              <enum>QAbstractItemView::SelectItems</enum>
             </property>
             <property name="verticalScrollMode">
              <enum>QAbstractItemView::ScrollPerPixel</enum>
             </property>
             <property name="horizontalScrollMode">
              <enum>QAbstractItemView::ScrollPerPixel</enum>
             </property>
             <property name="showGrid">
              <bool>false</bool>
             </property>
             <attribute name="horizontalHeaderHighlightSections">
              <bool>false</bool>
             </attribute>
             <attribute name="verticalHeaderHighlightSections">
              <bool>false</bool>
             </attribute>
            </widget>
           </item>
           <item row="0" column="0" colspan="9">
            <widget class="QLabel" name="label_3">
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Generated data</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="2" column="8">
            <widget class="QComboBox" name="cbViewType">
             <item>
              <property name="text">
               <string>Bit</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Hex</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>ASCII</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="btnNetworkSDRSend">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Send encoded data to your external application via TCP.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>Send via Network</string>
             </property>
             <property name="icon">
              <iconset theme="network-wired">
               <normaloff>.</normaloff>.</iconset>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="btnRfCatSend">
             <property name="text">
              <string>Send via RfCat</string>
             </property>
             <property name="icon">
              <iconset theme="network-wireless">
               <normaloff>.</normaloff>.</iconset>
             </property>
            </widget>
           </item>
          </layout>
          <zorder>cbViewType</zorder>
          <zorder>lViewType</zorder>
          <zorder>lEstimatedTime</zorder>
          <zorder>label_3</zorder>
          <zorder>tableMessages</zorder>
          <zorder>btnSave</zorder>
          <zorder>horizontalSpacer_3</zorder>
          <zorder>btnNetworkSDRSend</zorder>
          <zorder>btnRfCatSend</zorder>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GeneratorTreeView</class>
   <extends>QTreeView</extends>
   <header>urh.ui.views.GeneratorTreeView.h</header>
  </customwidget>
  <customwidget>
   <class>GeneratorTableView</class>
   <extends>QTableView</extends>
   <header>urh.ui.views.GeneratorTableView.h</header>
  </customwidget>
  <customwidget>
   <class>GeneratorListWidget</class>
   <extends>QListWidget</extends>
   <header>urh.ui.GeneratorListWidget.h</header>
  </customwidget>
  <customwidget>
   <class>GeneratorListView</class>
   <extends>QListView</extends>
   <header>urh.ui.views.GeneratorListView.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../urh.qrc"/>
 </resources>
 <connections/>
</ui>
